src/ngrams.cpp

```diff
@@ -1,12 +1,187 @@
+    cout << "Give me an input file and an 'N' value for groups" << endl;
+    cout << "of words, and I'll create random text for you." << endl;
+    cout << endl;
+
+    // Instatiate a vector for the word tokens form the text file and a map for the Ngrams
+    Vector<string> tokenVec;
+    Map<Vector<string>, Vector<string>> map;
+
+    // Ask for input text file from user to create the word tokens
+    getWordTokens(tokenVec);
 
+    // Get value of N which is least 2-gram to create the Ngrams map
+    int n = getValidInteger("Value of N? ", "N must be 2 or greater", 2);
+    cout << endl;
+    getNGramsMap(map, tokenVec, n);
+
+    // Repeatedly ask the user for the number of random words to be generated until they enter 0
+    // to quit
+    while (true) {
+        int numWordsToGen = getValidInteger("# of random words to generate (0 to quit)? ",
+                                            "Must be at least " + to_string(n) + " words.",
+                                            n,
+                                            true);
+        if (numWordsToGen == 0) {
+            break;
+        }
+        getRandomText(map, n, numWordsToGen);
+        cout << endl;
+    }
     cout << "Exiting." << endl;
     return 0;
 }
+
+/*
+ * Function: getWordTokens
+ * Usage: Prompts the user to get the reference text by typing the file name in order to get the
+ * word tokens from the file. Reprompts the user if an invalid file name is given.
+*/
+void getWordTokens(Vector<string> &tokenVec) {
+    string filename = promptUserForFile(
+                "Input file name? ","Unable to open that file. Try again.");
+    // Read the file into a string and instantiate a token scanner to get tokes from the text
+    string text = readEntireFile(filename);
+    TokenScanner scanner(text);
+    // Make scanner ignore white spaces and punctuation
+    scanner.ignoreWhitespace();
+    scanner.addWordCharacters("!\"#$%&'()*+,-./:;<=>?@[\]^_`{|}~");
+    // Put the tokens into a vector for reading
+    while (scanner.hasMoreTokens()) {
+        tokenVec.add(scanner.nextToken());
+    }
+}
+
+/*
+ * Function: getValidInteger
+ * Usage: Ensure user integer input is above a lower bound, other reprompt the user to re-enter.
+ * When existWhen0 is set to true, allows the user to exit the function when 0 is entered.
+*/
+int getValidInteger(const string &promptMessage,
+                    const string &repromptMessage,
+                    int lowerBound,
+                    bool exitWhen0) {
+    while (true) {
+        int userInput = getInteger(promptMessage);
+        if (exitWhen0) {
+            if (userInput == 0) {
+                return userInput;
+            }
+        }
+        if (userInput >= lowerBound) {
+            return userInput;
+        } else {
+            cout << repromptMessage << endl;
+        }
+    }
+}
+
+/*
+ * Function: getNGRamsMap
+ * Usage:Creates a map of the n-grams of words as the key and the following word as a value

```

Comment:
The values are actually a list of words (not a single word), so you can update this comment as per suggestion:

```suggestion
 * Usage:Creates a map of the n-grams of words as the key and the following words as a value

```